src/main/java/org/skriptlang/skript/lang/comparator/Comparators.java

```diff
@@ -0,0 +1,241 @@
+	 * @param firstType The first type for comparison.
+	 * @param secondType The second type for comparison.
+	 * @return A comparator capable of determine the {@link Relation} between two objects of 'firstType' and 'secondType'.
+	 * Will be null if no comparator capable of comparing two objects of 'firstType' and 'secondType' was found.
+	 * @param <Type1> The first type for comparison.
+	 * @param <Type2> The second type for comparison.
+	 * @param <CType1> The first type for any {@link ComparatorInfo}.
+	 * This is also used in organizing the conversion process of arguments (ex: 'Type1' to 'CType1' converter).
+	 * @param <CType2> The second type for any {@link ComparatorInfo}.
+	 * This is also used in organizing the conversion process of arguments (ex: 'Type2' to 'CType2' converter).
+	 */
+	@Nullable
+	@SuppressWarnings("unchecked")
+	private static <Type1, Type2, CType1, CType2> Comparator<Type1, Type2> getComparator_i(
+		Class<Type1> firstType,
+		Class<Type2> secondType
+	) {
+		
+		// Look for a perfect match
+		for (ComparatorInfo<?, ?> info : COMPARATORS) {
+			if (info.firstType.isAssignableFrom(firstType) && info.secondType.isAssignableFrom(secondType)) {
+				return (Comparator<Type1, Type2>) info.comparator;
+			}
+		}
+
+		// Try to match and create an InverseComparator
+		for (ComparatorInfo<?, ?> info : COMPARATORS) {
+			if (info.comparator.supportsInversion() && info.firstType.isAssignableFrom(secondType) && info.secondType.isAssignableFrom(firstType)) {
+				return new InverseComparator<>((Comparator<Type2, Type1>) info.comparator);
+			}
+		}
+
+		// Attempt converting one parameter
+		for (ComparatorInfo<?, ?> unknownInfo : COMPARATORS) {
+			ComparatorInfo<CType1, CType2> info = (ComparatorInfo<CType1, CType2>) unknownInfo;
+
+			if (info.firstType.isAssignableFrom(firstType)) { // Attempt to convert the second argument to the second comparator type
+				Converter<Type2, CType2> sc2 = Converters.getConverter(secondType, info.secondType);
+				if (sc2 != null)
+					return new ConvertedComparator<>(null, info.comparator, sc2);
+			}
+
+			if (info.secondType.isAssignableFrom(secondType)) { // Attempt to convert the first argument to the first comparator type
+				Converter<Type1, CType1> fc1 = Converters.getConverter(firstType, info.firstType);
+				if (fc1 != null)
+					return new ConvertedComparator<>(fc1, info.comparator, null);
+			}
+
+		}
+
+		// Attempt converting one parameter but with reversed types
+		for (ComparatorInfo<?, ?> unknownInfo : COMPARATORS) {
+			if (!unknownInfo.comparator.supportsInversion()) // Unsupported for reversing types
+				continue;
+
+			ComparatorInfo<CType1, CType2> info = (ComparatorInfo<CType1, CType2>) unknownInfo;
+
+			if (info.secondType.isAssignableFrom(firstType)) { // Attempt to convert the second argument to the first comparator type
+				Converter<Type2, CType1> sc1 = Converters.getConverter(secondType, info.firstType);
```

Comment:
You mixed it up, secondType is already assignable from firstType. Same below

Reply:
I don't think -

The Comparator's secondType is assignable from our firstType

Therefore, we are checking to determine if we can convert our secondType to the Comparator's firstType

Reply:
Yea I was a bit too quick there, I just misread it haha